src/main/kotlin/ar/edu/unahur/obj2/tareas/Tarea.kt

```diff
@@ -1,3 +1,34 @@
 package ar.edu.unahur.obj2.tareas
+import java.lang.Math.abs
 
-class Tarea
+interface Tarea {
+    fun nomina(): List<Trabajador>
+    fun horasNecesarias(): Int
+    fun costoTotalTarea(): Int
+}
+
+class TareaSimple :Tarea {
+    var costoInfraestructura = 0
+    var horasEstimadas = 0
+    val responsableDeTarea = mutableListOf<Trabajador>()
+    val empleados = mutableListOf<Trabajador>()
+    fun agregarResponsable(responsable: Trabajador) = responsableDeTarea.add(responsable)
+    fun agregarEmpleado(empleado:Trabajador) = empleados.add(empleado)
+    override fun horasNecesarias() = horasEstimadas / empleados.size
+    override fun nomina() : List<Trabajador> = empleados + responsableDeTarea
+    fun costoPorSueldosEmpleados():Int = empleados.sumBy { it.sueldo(this) }
+    fun costoPorSueldoResponsable():Int = responsableDeTarea.sumBy { it.sueldo(this) }
+    override fun costoTotalTarea(): Int = costoPorSueldosEmpleados() + costoPorSueldoResponsable() + costoInfraestructura
+}
+
+class TareaIntegracion : Tarea {
+    val responsableDeTarea= mutableListOf<Trabajador>()
+    val subtareas = mutableListOf<TareaSimple>()
+    fun agregarResponsable(responsable: Trabajador) = responsableDeTarea.add(responsable)
+    fun agregarSubtarea(tarea: TareaSimple) = subtareas.add(tarea)
+    override fun nomina() : List<Trabajador> = subtareas.map{ it.nomina() }.flatten() + responsableDeTarea
```

Comment:
Solo para que lo conozcan:
```suggestion
    override fun nomina() : List<Trabajador> = subtareas.flatMap{ it.nomina() } + responsableDeTarea
```